### DISCUSSION TITLE
Correct React 19 RC version selection

### DISCUSSION THREAD

**Discussion by karlismelderis-mckinsey** [renovatebot/renovate#32465] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github

### Please tell us more about your question or problem

React 19 RC has following versioning:
19.0.0-rc-69d4b800-20241021
19.0.0-rc-fb9a90fa48-20240614

`69d4b800` is commit
`20240614` is the date

At the moment Renovate is offering to "upgrade" `19.0.0-rc-69d4b800-20241021` with `19.0.0-rc-fb9a90fa48-20240614`

Is it possible to persuade Renovate to use date part to determine correct order of versions and ignore the commit? 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
It will need custom regex versioning as a workaround. Ideally they should have put the date first so that it sorts correctly using standard semver.

You can use 19.0.0 as the major.minor.patch while the date part can be `build`. It won't handle _duplicate_ releases on the same date (they will sort as the same value) but is that a problem?

**Reply by karlismelderis-mckinsey** [reply]
Can you please point to relevant part in documentation how regex versions works?

**Reply by rarkins** [reply, marked as answer]
https://docs.renovatebot.com/modules/versioning/#using-a-custom-regex-versioning-scheme

And lots of examples in past discussions

**Reply by karlismelderis-mckinsey** [reply]
💪
Thank you for the advice
I skimmed over Renovate documentation but as it's not a tiny tool I missed regex option in "versioning"

**Reply by rarkins** [reply]
Please post the solution you created too, in case it helps future users searching for the same thing

**Reply by karlismelderis-mckinsey** [reply]
we opted for this configuration:
```ts
    {
      "schedule": ["* * * * 2"],
      "groupName": "React",
      "matchPackageNames": ["react", "react-dom"],
      "versioning": "regex:^(?<major>\\d+).(?<minor>\\d+)\\.(?<patch>\\d+)-rc-[a-z0-9]{8}-(?<build>\\d{8})$"
    },
```
I suspect it stops updating react 18 to any latest react 18 version if there is project with anything but 19 RC
it's good enough for us but there is that limitation

**Reply by rarkins** [reply]
You could add `matchCurrentVersion` to ensure that it only does that when the current version also matches
